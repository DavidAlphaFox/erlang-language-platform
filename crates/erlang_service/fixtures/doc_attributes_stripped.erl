-module(doc_attributes_stripped).

-export([main/0]).

-if(OTP_RELEASE >= 27).
-moduledoc "
Check whether doc attributes are stripped from the AST and STUB sections.
".
-moduledoc #{
    authors => [<<"Metamate">>]
  }.

-doc "The main function".
-endif.

-spec main() -> ok.
main() ->
    ok.
