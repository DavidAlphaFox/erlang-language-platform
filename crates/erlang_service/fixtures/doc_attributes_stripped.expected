AST
{attribute,{0,0},file,{"fixtures/doc_attributes_stripped.erl",0}}.
{attribute,{0,32},module,doc_attributes_stripped}.
{attribute,{35,52},export,[{main,0}]}.
{attribute,{256,274},
           spec,
           {{main,0},
            [{type,{266,274},
                   'fun',
                   [{type,{266,274},product,[]},{atom,{272,274},ok}]}]}}.
{function,{276,292},main,0,[{clause,{276,292},[],[],[{atom,{290,292},ok}]}]}.
{eof,{294,294}}.


STUB
{attribute,{0,0},file,{"fixtures/doc_attributes_stripped.erl",0}}.
{attribute,{0,32},module,doc_attributes_stripped}.
{attribute,{35,52},export,[{main,0}]}.
{attribute,{256,274},
           spec,
           {{main,0},
            [{type,{266,274},
                   'fun',
                   [{type,{266,274},product,[]},{atom,{272,274},ok}]}]}}.


WARNINGS
[]

ERRORS
[]
